pilot/pkg/features/pilot.go

```diff
@@ -60,6 +60,9 @@ var (
 	PilotCertProvider = env.Register("PILOT_CERT_PROVIDER", constants.CertProviderIstiod,
 		"The provider of Pilot DNS certificate.").Get()
 
+	MetricsLocalhostAccessOnly = env.Register("PILOT_METRICS_LOCALHOST_ACCESS_ONLY", false,
```

Comment:
```
	PILOT_METRICS_LOCALHOST_ACCESS_ONLY
	PROXY_METRICS_LOCALHOST_ACCESS_ONLY
```

Do we have a use-case here for these to be separate? I would just as soon use

`METRICS_LOCALHOST_ACCESS_ONLY`

and define/document it once.

Reply:
In our use case right now we need to block for both proxy and control plane but I made it customized so that we can just block for proxy/Istiod if needed(maybe if control plane is not running in the same cluster and we want to block for proxy only). I am open to suggestions and can implement common flag also.

Reply:
At this time we only need one knob, so let's just add one knob.

We can split later if someone presents a compelling reason.

Reply:
Updated.